packages/@jsii/java-runtime/project/src/main/java/software/amazon/jsii/JsiiRuntime.java

```diff
@@ -178,33 +184,48 @@ protected void finalize() throws Throwable {
         }
     }
 
-    public void terminate() throws InterruptedException, IOException {
-        if (stderr != null) {
-            stderr.close();
-            stderr = null;
-        }
+    synchronized void terminate() {
+        try {
+            if (stdin != null) {
+                stdin.close();
+                stdin = null;
+            }
 
-        if (stdout != null) {
-            stdout.close();
-            stdout = null;
-        }
+            if (childProcess != null) {
```

Comment:
I feel like I am missing something... Who sends the termination signal to the child process? It seems like we are just waiting for 5 seconds and then killing it?

Reply:
The jsii Kernel interprets EOF on it's `STDIN` as a signal it should shut down, and obliges. We give it 5 seconds to clean up after itself, then dispose of it forcibly if it's still not done.

Reply:
Worth adding a comment in the relevant places because it's not obvious from the code